cpp/src/tests/snmg_pagerank/snmg_pagerank_test.cu

```diff
@@ -0,0 +1,280 @@
+  void verify_pr(gdf_column* col_pagerank, const MGPagerank_Usecase& param){
+    // Check vs golden data
+    if (param.result_file.length()>0)
+    {
+      int m = col_pagerank->size;
+      std::vector<val_t> calculated_res(m);
+      CUDA_RT_CALL(cudaMemcpy(&calculated_res[0],   col_pagerank->data, sizeof(val_t) * m, cudaMemcpyDeviceToHost));
+      std::sort(calculated_res.begin(), calculated_res.end());
+      FILE* fpin = fopen(param.result_file.c_str(),"rb");
+      ASSERT_TRUE(fpin != NULL) << " Cannot read file with reference data: " << param.result_file << std::endl;
+      std::vector<val_t> expected_res(m);
+      ASSERT_EQ(read_binary_vector(fpin, m, expected_res), 0);
+      fclose(fpin);
+      val_t err;
+      int n_err = 0;
+      for (int i = 0; i < m; i++) {
+          //if(i > (m-10))
+          //  std::cout << expected_res[i] << " " << calculated_res[i] <<std::endl;
+          err = fabs(expected_res[i] - calculated_res[i]);
+          if (err> 1e-5)
+          {
+              n_err++;
+          }
+      }
+      if (n_err) {
+          //EXPECT_NEAR(tot_err/n_err, cugraph_Const<T>::tol, cugraph_Const<T>::tol*9.99); // Network x used n*1e-10 for precision
+          EXPECT_LE(n_err, 0.001*m); // we tolerate 0.1% of values with a litte difference
+          //printf("number of incorrect entries: %d\n", n_err);
+          //if (n_err > 0.001*m)
+          //{
+          //  eq(calculated_res,expected_res);
+          //}
+      }
+    }
+  }
+  template <typename idx_t,typename val_t>
+  void run_current_test(const MGPagerank_Usecase& param) {
+     const ::testing::TestInfo* const test_info =::testing::UnitTest::GetInstance()->current_test_info();
+     std::stringstream ss; 
+     std::string test_id = std::string(test_info->test_case_name()) + std::string(".") + std::string(test_info->name()) + std::string("_") + getFileName(param.matrix_file)+ std::string("_") + ss.str().c_str();
+
+     int m, k, nnz, n_gpus, max_iter=50;
+     val_t alpha = 0.85;
+     MM_typecode mc;
+
+     double t;
+
+     FILE* fpin = fopen(param.matrix_file.c_str(),"r");
+     
+     ASSERT_EQ(mm_properties<int>(fpin, 1, &mc, &m, &k, &nnz),0) << "could not read Matrix Market file properties"<< "\n";
+     ASSERT_TRUE(mm_is_matrix(mc));
+     ASSERT_TRUE(mm_is_coordinate(mc));
+     ASSERT_FALSE(mm_is_complex(mc));
+     ASSERT_FALSE(mm_is_skew(mc));
+     
+     // Allocate memory on host
+     std::vector<idx_t> cooRowInd(nnz), cooColInd(nnz), csrColInd(nnz), csrRowPtr(m+1);
+     std::vector<val_t> cooVal(nnz), csrVal(nnz), pagerank_h(m, 1.0/m), y_h(m, 0.0), y_ref(m, 0.0);
+
+     // Read
+     ASSERT_EQ( (mm_to_coo<int,val_t>(fpin, 1, nnz, &cooRowInd[0], &cooColInd[0], NULL, NULL)) , 0)<< "could not read matrix data"<< "\n";
+     ASSERT_EQ(fclose(fpin),0);
+     //ASSERT_EQ( (coo_to_csr<int,val_t> (m, m, nnz, &cooRowInd[0],  &cooColInd[0], NULL, NULL, &csrRowPtr[0], NULL, NULL, NULL)), 0) << "could not covert COO to CSR "<< "\n";
```

Comment:
Can this be removed?

Reply:
Agreed but will be addressed in the next PR. This file is WIP and will be finalized along with the rest of the Pagernak solver. I'd rather do all the cleaning at once.